### PR TITLE
[WIP]: Add workspace name to title bar

### PR DESCRIPTION
**Pull request by goanpeca**
## References

https://github.com/jupyterlab/jupyterlab/issues/6946

## User-facing changes

title

Plus one with emoji in name

image

### CODE REVIEW

**Comment by echarles** [comment]
Thx @goanpeca Overall code looks good. Can you resolve the conflict?

On https://github.com/jupyterlab/jupyterlab/issues/6946#issuecomment-675804557 you say `I added + N to denote all the items.` I have followed the animated gif, but as a user, I am a bit confused: e.g. there are 4 tabs in the main area, and the title shows `+ 3`.

I am not a big fan of `+` as separator, but it may just be me, maybe @isabela-pf has thoughts on that?

File: dev_mode/webpack.config.js [hunk 1]
```diff
@@ -222,7 +222,7 @@ module.exports = [
     },
     output: {
       path: plib.resolve(buildDir),
-      publicPath: 'static/lab/',
+      publicPath: '/static/lab/',
```
Comment (echarles) [inline dev_mode/webpack.config.js:225]: Is that additional `/` needed? It maybe a rest of the time when the workspace loading were temporary no more working?

**Review by echarles — CHANGES_REQUESTED** [review]

**Comment by goanpeca** [comment]
Hi @echarles, yes I will fix the conflict.

>you say I added + N to denote all the items. I have followed the animated gif, but as a user, I am a bit confused: e.g. there are 4 tabs in the main area, and the title shows + 3.

Absolutely I need some UI/UX help from @isabela-pf :) !

**Comment by goanpeca** [comment]
@echarles 

>bit confused: e.g. there are 4 tabs in the main area, and the title shows + 3.

So it is the current thing plus the rest, so 4 in total would be.  Something + 3  = 4. Yeah I agree the `+` is weird, but need something for ilustartion.

**Comment by echarles** [comment]
> Something + 3 = 4.

ah ah, was reading `+` like a `separator`, but it would be more like a `sum`.

File: dev_mode/webpack.config.js (hunk 1 above)
Comment (goanpeca) [reply dev_mode/webpack.config.js:225]: Yes it is possible, will rebase!

**Comment by isabela-pf** [comment]
I'm not sure the "+N" approach is the most clear, but I do think it might be the most succinct option. It would be a little clearer even just having it say "+N more," but it does push the JupyterLab name out of the visible part of the tab. I've also seen similar UI use "Document Name (N more)" since the + could be confusing if it isn't read as plus/and.

I haven't decided what approach I like better (both seem to read fine on a screen reader; I wasn't sure if the + would read in a way that didn't make sense), but I didn't want to make you wait longer on my thoughts.

**Comment by ellisonbg** [comment]
Not sure about putting the active document name of the browser tab, seems overly contextual and would change over time. But I think it would be helpful to have the count of open documents/widgets.

**Comment by isabela-pf** [comment]
> Not sure about putting the active document name of the browser tab, seems overly contextual and would change over time. But I think it would be helpful to have the count of open documents/widgets.

I think that's a good point, but it still seems like the most useful information. Are you thinking something like "# files - JupyterLab"? Or when there is no workspace set it just defaults to plain "JupyterLab?" 

If someone isn't using workspaces but is having multiple JupyterLabs open (the way that I think titles would be most useful) it seems like what files are open is the key information. Do you think it would be less jarring to have a single file (maybe the first opened?) be in the title rather than it changing with active files?

**Comment by jess-x** [comment]
Hi @isabela-pf ,
I played with the naming of title bar for workspaces, and changed to `+N` to `(name of workspace: number of documents/widgets)`. I also changed the `default` workspace to the active document/widget name. Would love your opinion on this.  
Screen Shot 2021-03-24 at 6 02 06 PM

**Comment by goanpeca** [comment]
Closing in favor of https://github.com/jupyterlab/jupyterlab/pull/10002/files

**Pull request closed without merging**